avm/res/network/dnsresolver/main.bicep

```
@@ -0,0 +1,168 @@
+metadata name = 'DNS Resolvers'
+metadata description = 'This module deploys a DNS Resolver.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Private DNS Resolver.')
+@minLength(1)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. ResourceId of the virtual network to attach the Private DNS Resolver to.')
+param virtualNetworkId string
```

```suggestion
param virtualNetworkResourceId string
```
requires references to be updated too